pytorch_lightning/trainer/deprecated_api.py

```diff
@@ -121,7 +121,7 @@ def show_progress_bar(self):
         """Back compatibility, will be removed in v0.9.0"""
         rank_zero_warn("Argument `show_progress_bar` is now set by `progress_bar_refresh_rate` since v0.7.2"
                        " and this method will be removed in v0.9.0", DeprecationWarning)
-        return self.progress_bar_refresh_rate >= 1
+        return self.progress_bar_callback and self.progress_bar_callback.refresh_rate >= 1
```

Comment:
How is this implicit conversion of object instances to bool handled? Is every object (except empty containers) casted to True?

Reply:
yes exactly, empty containers like lists or dicts have `__len__` defined as 0 and so they evaluate to False. If the object here also had the len defined, it would evaluate based on that, but otherwise defaults to `True`.